### PR TITLE
Add ETag Support

### LINKED ISSUE #12045: Etag for API routes
# Feature request

API routes should support the [Etag](https://tools.ietf.org/html/rfc7232#section-2.3) spec to enable ETag-based caching.

## Is your feature request related to a problem? Please describe.

Some of my API routes return a not-so-tiny response, which I'd love to be cached using ETag.

## Describe the solution you'd like

API routes should adhere to ETag spec.

## Describe alternatives you've considered

Rolling my own etag support.

### PR DESCRIPTION
**Pull request by Skn0tt**
Closes #12045 

This PR adds support for [etags](https://tools.ietf.org/html/rfc7232#section-3.2) to Next.js' API routes, which will improve user experience and decrease network traffic by enabling usage of etag-based caching.

### CODE REVIEW

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Server Mode</strong> (Decrease detected ✓)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | zeit/next.js canary  | Skn0tt/next.js fix-12045 | Change |
| - | - | - | - |
| buildDuration | 14.1s | 13.5s | -559ms |
| nodeModulesSize | 58.7 MB | 58.7 MB | ⚠️ +11.7 kB |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall decrease ⚠️</summary>

|  | zeit/next.js canary  | Skn0tt/next.js fix-12045 | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 2.48 | 2.312 | -0.17 |
| / avg req/sec | 1008 | 1081.42 | ⚠️ +73.42 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.534 | 1.622 | ⚠️ +0.09 |
| /error-in-render avg req/sec | 1629.67 | 1541.28 | -88.39 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | zeit/next.js canary  | Skn0tt/next.js fix-12045 | Change |
| - | - | - | - |
| main-HASH.js gzip | 6.68 kB | 6.68 kB | ✓ |
| webpack-HASH.js gzip | 746 B | 746 B | ✓ |
| de003c3a9d30..e49d.js gzip | 10.4 kB | 10.4 kB | ✓ |
| framework.HASH.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 57 kB | 57 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | zeit/next.js canary  | Skn0tt/next.js fix-12045 | Change |
| - | - | - | - |
| main-HASH.module.js gzip | 5.72 kB | 5.72 kB | ✓ |
| webpack-HASH..dule.js gzip | 746 B | 746 B | ✓ |
| de003c3a9d30..dule.js gzip | 6.85 kB | 6.85 kB | ✓ |
| framework.HA..dule.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 52.5 kB | 52.5 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | zeit/next.js canary  | Skn0tt/next.js fix-12045 | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 26.3 kB | 26.3 kB | ✓ |
| Overall change | 26.3 kB | 26.3 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | zeit/next.js canary  | Skn0tt/next.js fix-12045 | Change |
| - | - | - | - |
| _app.js gzip | 1.26 kB | 1.26 kB | ✓ |
| _error.js gzip | 3.29 kB | 3.29 kB | ✓ |
| hooks.js gzip | 880 B | 880 B | ✓ |
| index.js gzip | 222 B | 222 B | ✓ |
| link.js gzip | 2.06 kB | 2.06 kB | ✓ |
| routerDirect.js gzip | 280 B | 280 B | ✓ |
| withRouter.js gzip | 278 B | 278 B | ✓ |
| Overall change | 8.26 kB | 8.26 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages Modern</strong></summary>

|  | zeit/next.js canary  | Skn0tt/next.js fix-12045 | Change |
| - | - | - | - |
| _app.module.js gzip | 599 B | 599 B | ✓ |
| _error.module.js gzip | 2.09 kB | 2.09 kB | ✓ |
| hooks.module.js gzip | 383 B | 383 B | ✓ |
| index.module.js gzip | 223 B | 223 B | ✓ |
| link.module.js gzip | 1.52 kB | 1.52 kB | ✓ |
| routerDirect..dule.js gzip | 279 B | 279 B | ✓ |
| withRouter.m..dule.js gzip | 278 B | 278 B | ✓ |
| Overall change | 5.37 kB | 5.37 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | zeit/next.js canary  | Skn0tt/next.js fix-12045 | Change |
| - | - | - | - |
| _buildManifest.js gzip | 61 B | 61 B | ✓ |
| _buildManife..dule.js gzip | 61 B | 61 B | ✓ |
| Overall change | 122 B | 122 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | zeit/next.js canary  | Skn0tt/next.js fix-12045 | Change |
| - | - | - | - |
| index.html gzip | 927 B | 927 B | ✓ |
| link.html gzip | 937 B | 937 B | ✓ |
| withRouter.html gzip | 925 B | 925 B | ✓ |
| Overall change | 2.79 kB | 2.79 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Serverless Mode</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | zeit/next.js canary  | Skn0tt/next.js fix-12045 | Change |
| - | - | - | - |
| buildDuration | 15.5s | 15.1s | -387ms |
| nodeModulesSize | 58.7 MB | 58.7 MB | ⚠️ +11.7 kB |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | zeit/next.js canary  | Skn0tt/next.js fix-12045 | Change |
| - | - | - | - |
| main-HASH.js gzip | 6.68 kB | 6.68 kB | ✓ |
| webpack-HASH.js gzip | 746 B | 746 B | ✓ |
| de003c3a9d30..e49d.js gzip | 10.4 kB | 10.4 kB | ✓ |
| framework.HASH.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 57 kB | 57 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | zeit/next.js canary  | Skn0tt/next.js fix-12045 | Change |
| - | - | - | - |
| main-HASH.module.js gzip | 5.72 kB | 5.72 kB | ✓ |
| webpack-HASH..dule.js gzip | 746 B | 746 B | ✓ |
| de003c3a9d30..dule.js gzip | 6.85 kB | 6.85 kB | ✓ |
| framework.HA..dule.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 52.5 kB | 52.5 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | zeit/next.js canary  | Skn0tt/next.js fix-12045 | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 26.3 kB | 26.3 kB | ✓ |
| Overall change | 26.3 kB | 26.3 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | zeit/next.js canary  | Skn0tt/next.js fix-12045 | Change |
| - | - | - | - |
| _app.js gzip | 1.26 kB | 1.26 kB | ✓ |
| _error.js gzip | 3.29 kB | 3.29 kB | ✓ |
| hooks.js gzip | 880 B | 880 B | ✓ |
| index.js gzip | 222 B | 222 B | ✓ |
| link.js gzip | 2.06 kB | 2.06 kB | ✓ |
| routerDirect.js gzip | 280 B | 280 B | ✓ |
| withRouter.js gzip | 278 B | 278 B | ✓ |
| Overall change | 8.26 kB | 8.26 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages Modern</strong></summary>

|  | zeit/next.js canary  | Skn0tt/next.js fix-12045 | Change |
| - | - | - | - |
| _app.module.js gzip | 599 B | 599 B | ✓ |
| _error.module.js gzip | 2.09 kB | 2.09 kB | ✓ |
| hooks.module.js gzip | 383 B | 383 B | ✓ |
| index.module.js gzip | 223 B | 223 B | ✓ |
| link.module.js gzip | 1.52 kB | 1.52 kB | ✓ |
| routerDirect..dule.js gzip | 279 B | 279 B | ✓ |
| withRouter.m..dule.js gzip | 278 B | 278 B | ✓ |
| Overall change | 5.37 kB | 5.37 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | zeit/next.js canary  | Skn0tt/next.js fix-12045 | Change |
| - | - | - | - |
| _buildManifest.js gzip | 61 B | 61 B | ✓ |
| _buildManife..dule.js gzip | 61 B | 61 B | ✓ |
| Overall change | 122 B | 122 B | ✓ |

</details>

<details>
<summary><strong>Serverless bundles</strong> Overall increase ⚠️</summary>

|  | zeit/next.js canary  | Skn0tt/next.js fix-12045 | Change |
| - | - | - | - |
| _error.js | 850 kB | 853 kB | ⚠️ +3.28 kB |
| 404.html | 4.18 kB | 4.18 kB | ✓ |
| hooks.html | 3.81 kB | 3.81 kB | ✓ |
| index.js | 850 kB | 853 kB | ⚠️ +3.28 kB |
| link.js | 887 kB | 891 kB | ⚠️ +3.28 kB |
| routerDirect.js | 880 kB | 883 kB | ⚠️ +3.28 kB |
| withRouter.js | 880 kB | 883 kB | ⚠️ +3.28 kB |
| Overall change | 4.35 MB | 4.37 MB | ⚠️ +16.4 kB |

</details>

</details>

**Comment by ijjk** [comment]
## Failing test suites 

**test/integration/api-support/test/index.test.js** 
- API routes > Server support > should support etag spec
- API routes > Serverless support > should support etag spec
- API routes > dev support > should support etag spec

<details>
<summary>Expand output</summary>

  ● API routes › dev support › should support etag spec

    expect(received).toMatch(expected)

    Matcher error: received value must be a string

    Received has value: null

      164 |     const etag = response.headers.get('etag')
      165 | 
    > 166 |     expect(etag).toMatch(/^W\/.*$/)
          |                  ^
      167 | 
      168 |     const unmodifiedResponse = await fetchViaHTTP(appPort, '/api/blog', null, {
      169 |       headers: { 'If-None-Match': etag },

      at Object.<anonymous> (integration/api-support/test/index.test.js:166:18)

  ● API routes › Server support › should support etag spec

    expect(received).toMatch(expected)

    Matcher error: received value must be a string

    Received has value: null

      164 |     const etag = response.headers.get('etag')
      165 | 
    > 166 |     expect(etag).toMatch(/^W\/.*$/)
          |                  ^
      167 | 
      168 |     const unmodifiedResponse = await fetchViaHTTP(appPort, '/api/blog', null, {
      169 |       headers: { 'If-None-Match': etag },

      at Object.<anonymous> (integration/api-support/test/index.test.js:166:18)

  ● API routes › Serverless support › should support etag spec

    expect(received).toMatch(expected)

    Matcher error: received value must be a string

    Received has value: null

      164 |     const etag = response.headers.get('etag')
      165 | 
    > 166 |     expect(etag).toMatch(/^W\/.*$/)
          |                  ^
      167 | 
      168 |     const unmodifiedResponse = await fetchViaHTTP(appPort, '/api/blog', null, {
      169 |       headers: { 'If-None-Match': etag },

      at Object.<anonymous> (integration/api-support/test/index.test.js:166:18)

</details>

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Server Mode</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | zeit/next.js canary  | Skn0tt/next.js fix-12045 | Change |
| - | - | - | - |
| buildDuration | 13s | 12.8s | -130ms |
| nodeModulesSize | 58.7 MB | 58.7 MB | ⚠️ +11.7 kB |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall increase ✓</summary>

|  | zeit/next.js canary  | Skn0tt/next.js fix-12045 | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 2.213 | 2.192 | -0.02 |
| / avg req/sec | 1129.91 | 1140.61 | ⚠️ +10.7 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.452 | 1.427 | -0.02 |
| /error-in-render avg req/sec | 1721.34 | 1751.36 | ⚠️ +30.02 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | zeit/next.js canary  | Skn0tt/next.js fix-12045 | Change |
| - | - | - | - |
| main-HASH.js gzip | 6.68 kB | 6.68 kB | ✓ |
| webpack-HASH.js gzip | 746 B | 746 B | ✓ |
| de003c3a9d30..e49d.js gzip | 10.4 kB | 10.4 kB | ✓ |
| framework.HASH.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 57 kB | 57 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | zeit/next.js canary  | Skn0tt/next.js fix-12045 | Change |
| - | - | - | - |
| main-HASH.module.js gzip | 5.72 kB | 5.72 kB | ✓ |
| webpack-HASH..dule.js gzip | 746 B | 746 B | ✓ |
| de003c3a9d30..dule.js gzip | 6.85 kB | 6.85 kB | ✓ |
| framework.HA..dule.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 52.5 kB | 52.5 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | zeit/next.js canary  | Skn0tt/next.js fix-12045 | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 26.3 kB | 26.3 kB | ✓ |
| Overall change | 26.3 kB | 26.3 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | zeit/next.js canary  | Skn0tt/next.js fix-12045 | Change |
| - | - | - | - |
| _app.js gzip | 1.26 kB | 1.26 kB | ✓ |
| _error.js gzip | 3.29 kB | 3.29 kB | ✓ |
| hooks.js gzip | 880 B | 880 B | ✓ |
| index.js gzip | 222 B | 222 B | ✓ |
| link.js gzip | 2.06 kB | 2.06 kB | ✓ |
| routerDirect.js gzip | 280 B | 280 B | ✓ |
| withRouter.js gzip | 278 B | 278 B | ✓ |
| Overall change | 8.26 kB | 8.26 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages Modern</strong></summary>

|  | zeit/next.js canary  | Skn0tt/next.js fix-12045 | Change |
| - | - | - | - |
| _app.module.js gzip | 599 B | 599 B | ✓ |
| _error.module.js gzip | 2.09 kB | 2.09 kB | ✓ |
| hooks.module.js gzip | 383 B | 383 B | ✓ |
| index.module.js gzip | 223 B | 223 B | ✓ |
| link.module.js gzip | 1.52 kB | 1.52 kB | ✓ |
| routerDirect..dule.js gzip | 279 B | 279 B | ✓ |
| withRouter.m..dule.js gzip | 278 B | 278 B | ✓ |
| Overall change | 5.37 kB | 5.37 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | zeit/next.js canary  | Skn0tt/next.js fix-12045 | Change |
| - | - | - | - |
| _buildManifest.js gzip | 61 B | 61 B | ✓ |
| _buildManife..dule.js gzip | 61 B | 61 B | ✓ |
| Overall change | 122 B | 122 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | zeit/next.js canary  | Skn0tt/next.js fix-12045 | Change |
| - | - | - | - |
| index.html gzip | 927 B | 927 B | ✓ |
| link.html gzip | 937 B | 937 B | ✓ |
| withRouter.html gzip | 925 B | 925 B | ✓ |
| Overall change | 2.79 kB | 2.79 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Serverless Mode</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | zeit/next.js canary  | Skn0tt/next.js fix-12045 | Change |
| - | - | - | - |
| buildDuration | 13.6s | 14s | ⚠️ +411ms |
| nodeModulesSize | 58.7 MB | 58.7 MB | ⚠️ +11.7 kB |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | zeit/next.js canary  | Skn0tt/next.js fix-12045 | Change |
| - | - | - | - |
| main-HASH.js gzip | 6.68 kB | 6.68 kB | ✓ |
| webpack-HASH.js gzip | 746 B | 746 B | ✓ |
| de003c3a9d30..e49d.js gzip | 10.4 kB | 10.4 kB | ✓ |
| framework.HASH.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 57 kB | 57 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | zeit/next.js canary  | Skn0tt/next.js fix-12045 | Change |
| - | - | - | - |
| main-HASH.module.js gzip | 5.72 kB | 5.72 kB | ✓ |
| webpack-HASH..dule.js gzip | 746 B | 746 B | ✓ |
| de003c3a9d30..dule.js gzip | 6.85 kB | 6.85 kB | ✓ |
| framework.HA..dule.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 52.5 kB | 52.5 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | zeit/next.js canary  | Skn0tt/next.js fix-12045 | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 26.3 kB | 26.3 kB | ✓ |
| Overall change | 26.3 kB | 26.3 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | zeit/next.js canary  | Skn0tt/next.js fix-12045 | Change |
| - | - | - | - |
| _app.js gzip | 1.26 kB | 1.26 kB | ✓ |
| _error.js gzip | 3.29 kB | 3.29 kB | ✓ |
| hooks.js gzip | 880 B | 880 B | ✓ |
| index.js gzip | 222 B | 222 B | ✓ |
| link.js gzip | 2.06 kB | 2.06 kB | ✓ |
| routerDirect.js gzip | 280 B | 280 B | ✓ |
| withRouter.js gzip | 278 B | 278 B | ✓ |
| Overall change | 8.26 kB | 8.26 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages Modern</strong></summary>

|  | zeit/next.js canary  | Skn0tt/next.js fix-12045 | Change |
| - | - | - | - |
| _app.module.js gzip | 599 B | 599 B | ✓ |
| _error.module.js gzip | 2.09 kB | 2.09 kB | ✓ |
| hooks.module.js gzip | 383 B | 383 B | ✓ |
| index.module.js gzip | 223 B | 223 B | ✓ |
| link.module.js gzip | 1.52 kB | 1.52 kB | ✓ |
| routerDirect..dule.js gzip | 279 B | 279 B | ✓ |
| withRouter.m..dule.js gzip | 278 B | 278 B | ✓ |
| Overall change | 5.37 kB | 5.37 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | zeit/next.js canary  | Skn0tt/next.js fix-12045 | Change |
| - | - | - | - |
| _buildManifest.js gzip | 61 B | 61 B | ✓ |
| _buildManife..dule.js gzip | 61 B | 61 B | ✓ |
| Overall change | 122 B | 122 B | ✓ |

</details>

<details>
<summary><strong>Serverless bundles</strong> Overall increase ⚠️</summary>

|  | zeit/next.js canary  | Skn0tt/next.js fix-12045 | Change |
| - | - | - | - |
| _error.js | 850 kB | 854 kB | ⚠️ +3.32 kB |
| 404.html | 4.18 kB | 4.18 kB | ✓ |
| hooks.html | 3.81 kB | 3.81 kB | ✓ |
| index.js | 851 kB | 854 kB | ⚠️ +3.32 kB |
| link.js | 888 kB | 891 kB | ⚠️ +3.32 kB |
| routerDirect.js | 880 kB | 884 kB | ⚠️ +3.32 kB |
| withRouter.js | 880 kB | 884 kB | ⚠️ +3.32 kB |
| Overall change | 4.36 MB | 4.37 MB | ⚠️ +16.6 kB |

</details>

</details>

File: packages/next/next-server/server/api-utils.ts [hunk 1]
```diff
@@ -197,50 +198,75 @@ export function sendStatusCode(res: NextApiResponse, statusCode: number) {
...
 }
 
+/**
+ * Try to match ETag according to RFC7232
+ * @implements https://tools.ietf.org/html/rfc7232#section-3.2
+ * @param req request object
+ * @param res response object
+ * @param body of response
+ */
+function matchETag(
```
Comment (Timer) [inline packages/next/next-server/server/api-utils.ts:208]: We have the `fresh` package installed which should handle all of this.

File: packages/next/package.json [hunk 2]
```diff
@@ -84,6 +84,7 @@
     "cacache": "13.0.1",
     "chokidar": "2.1.8",
     "css-loader": "3.5.3",
+    "etag": "1.8.1",
```
Comment (Timer) [inline packages/next/package.json:87]:
```suggestion
```

File: packages/next/next-server/server/api-utils.ts (hunk 1 above)
Comment (Timer) [reply packages/next/next-server/server/api-utils.ts:208]: See `packages/next/next-server/server/send-html.ts`.

**Review by Timer — CHANGES_REQUESTED** [review]

File: packages/next/next-server/server/api-utils.ts [hunk 3]
```diff
@@ -8,6 +8,7 @@ import { isResSent, NextApiRequest, NextApiResponse } from '../lib/utils'
 import { decryptWithSecret, encryptWithSecret } from './crypto-utils'
 import { interopDefault } from './load-components'
 import { Params } from './router'
+import computeEtag from 'etag'
```
[the remark this answers is not in the document]
Comment (Skn0tt) [reply packages/next/next-server/server/api-utils.ts]: Fixed in https://github.com/vercel/next.js/pull/12802/commits/7e854211b068a2536c3c17b0639b29032e324245

File: packages/next/package.json (hunk 2 above)
Comment (Skn0tt) [reply packages/next/package.json]: Fixed in https://github.com/vercel/next.js/pull/12802/commits/7e854211b068a2536c3c17b0639b29032e324245

File: packages/next/next-server/server/api-utils.ts (hunk 1 above)
Comment (Skn0tt) [reply packages/next/next-server/server/api-utils.ts:209]: Fresh doesn't deal with all of this, but it replaces the `etag === req.headers['if-none-match']` part. See 5f9f336

Comment (Skn0tt) [reply packages/next/next-server/server/api-utils.ts:209]: Is that what you had in mind?

Comment (Timer) [reply packages/next/next-server/server/api-utils.ts:215]: Yes, this LGTM.

**Review by Timer — DISMISSED** [review]
Thank you!

**Review by Timer — APPROVED** [review]

**Comment by timneutkens** [comment]
Awesome PR @Skn0tt!

**Pull request merged**